lib/fined.js

```diff
@@ -0,0 +1,141 @@
+const fs = require('fs');
+const path = require('path');
+
+const isArray = require('lodash.isarray');
+const isString = require('lodash.isstring');
+const isPlainObject = require('lodash.isplainobject');
+const isEmpty = require('lodash.isempty');
+const mapValues = require('lodash.mapvalues');
+const pick = require('lodash.pick');
+
+const expandTilde = require('expand-tilde');
+const parsePath = require('parse-filepath');
+
+
+function search(options, extensions, cwd) {
```

Comment:
Put yet *another* way, you should never embed iteration into a primitive. Make your primitive act on a single atomic unit. Having done that you can use any control flow you like to call the primitive multiple times.

In some sense this is what you've done:
```js
function doubler(input) {
  if (Array.isArray(input)) { 
    return input.map(x => x * 2);
  }
  return input * 2;
}
doubler(1);
doubler([1,2,3]);
```

...when this is much easier to reason about / unit test / etc:
```
function doubler(input) {
  return input * 2;
}
doubler(1);
[1,2,3].map(doubler);
```